Kernel/FileSystem/TmpFS.cpp

```diff
@@ -0,0 +1,295 @@
+#include <Kernel/FileSystem/TmpFS.h>
+#include <Kernel/Process.h>
+#include <Kernel/Thread.h>
+
+NonnullRefPtr<TmpFS> TmpFS::create()
+{
+    return adopt(*new TmpFS);
+}
+
+TmpFS::TmpFS()
+{
+}
+
+TmpFS::~TmpFS()
+{
+}
+
+bool TmpFS::initialize()
+{
+    m_root_inode = TmpFSInode::create_root(*this);
+    return true;
+}
+
+InodeIdentifier TmpFS::root_inode() const
+{
+    ASSERT(!m_root_inode.is_null());
+    return m_root_inode->identifier();
+}
+
+void TmpFS::register_inode(NonnullRefPtr<TmpFSInode> inode)
+{
+    LOCKER(m_lock);
+    ASSERT(inode->identifier().fsid() == fsid());
+
+    unsigned index = inode->identifier().index();
+    m_inodes.set(index, move(inode));
+}
+
+void TmpFS::unregister_inode(InodeIdentifier identifier)
+{
+    LOCKER(m_lock);
+    ASSERT(identifier.fsid() == fsid());
+
+    m_inodes.remove(identifier.index());
+}
+
+unsigned TmpFS::next_inode_index()
+{
+    LOCKER(m_lock);
+
+    return m_next_inode_index++;
+}
+
+RefPtr<Inode> TmpFS::get_inode(InodeIdentifier identifier) const
+{
+    LOCKER(m_lock);
+    ASSERT(identifier.fsid() == fsid());
+
+    auto it = m_inodes.find(identifier.index());
```

Comment:
This can also be written as
```cpp
return m_inodes.get(identifier.index()).value_or(nullptr);
```

Reply:
That's a nice shortcut, but it doesn't seem to work here; `m_inodes` stores `NonullRefPtr<Inode>`s, so we cannot use `nullptr` as a default value; the conversion to `RefPtr<Inode>` is happening after we get it from the map.